dedalus/core/basis.py

```diff
@@ -5534,9 +5534,7 @@ def cfl_spacing(self, r=None):
         if basis.Lmax == 0:
             s2_spacing[:] = np.inf
         else:
-            for i in range(len(s2_spacing.shape)):
-                if s2_spacing.shape[i] > 1:
-                    s2_spacing[i, :] = r / np.sqrt(basis.Lmax*(basis.Lmax + 1))
+            s2_spacing[:] = r / np.sqrt(basis.Lmax*(basis.Lmax + 1))
```

Comment:
I think we still want the check on the spacing shapes, right? To catch the case where Nphi=1 (so mmax=0 but Lmax > 0)?

Reply:
I don't think we do. But this may reflect a different design choice that should be made here.

We still want to preserve the r / sqrt(L*(L+1)) behavior of the CFL in the theta direction, and this current formulation does that in both BallBases and ShellBases. If s2_spacing were setting something that depended on phi and theta, then we'd want to be more careful, but since it's just setting a scalar value I think the only conditioning necessary is when Lmax = 0.

Reply:
Ah, right, because we're using the magnitude of the horizontal velocity to compute the CFL, not the individual components. That will just mean that there's a CFL restriction associated with zonal velocities in axisymmetric problems. Maybe not a big deal for now, but maybe that's something we eventually want to fix?

Reply:
Ah! Good point, I see what you're saying and we should probably address it here. I think this calls for a minor redesign (s2_spacing should have a (ntheta, nphi) shape)). I'll think about that a bit and update it within a couple hours, hopefully.

Reply:
We have to think about it a bit carefully. Say you have an axisymmetric (Nphi=1) velocity field V that you use to advect a non-axisymmetric (Nphi > 1) scalar T. In that case there will still be an azimuthal CFL limit, but we can't infer the right grid spacing to use for it from the bases of the velocity field itself...

Reply:
Hmm. Ok, on second thought, this is feeling like it's beyond the scope of this PR (which is to make it so ntheta = 1 balls/shells don't crash on initialization), and that's more of a new feature? Maybe we should open a separate issue?

Reply:
Yep! Merge away, CFL seems to be initializing in both parallel and serial, for balls and shells.